### ISSUE TITLE
signal.sh SIGINT trapping fails on BSD/macOS

### ISSUE THREAD

**Issue by krader1961** [att/ast#658]
On macOS and other BSD based systems the portion of *signal.sh* which tests SIGINT trapping fails with errors like those below. This has been true since I started contributing a year ago. But now that we've fixed nearly all the tests and are no longer disabling the *signal.sh* unit test it's time to explore what is going on with this specific test on non-Linux (and presumably SysV) systems.

```
<E> sigtst1[12]: .t. expected message not received
<E> expect: |int1|
<E> actual: ||
<E> sigtst1[21]: .t. int_trap should have run only once,but ran 2 times
<W> sigtst1[-1]: error_count = 2
<E> sigtst1[12]: .td expected message not received
<E> expect: |int1|
<E> actual: ||
<E> sigtst1[21]: .td int_trap should have run only once,but ran 2 times
<W> sigtst1[-1]: error_count = 2
<E> signal[146]: sigint .t. test failed
<E> expect: |1-main 2-main 3-main SIGINT 3-intr 2-intr 1-intr 1-0258|
<E> actual: |1-main 2-main 3-main SIGINT 3-intr 2-intr 1-intr 1-intr 1-0258|
<E> signal[146]: sigint .td test failed
<E> expect: |1-main 2-main 3-main SIGINT 3-intr 2-intr 1-intr 1-0258|
<E> actual: |1-main 2-main 3-main SIGINT 3-intr 2-intr 1-intr 1-intr 1-0258|
```

**Comment by krader1961** [comment]
I fixed this long ago when I rewrote the *signal.sh* unit test among other changes.

**Comment by krader1961** [comment]
When I wrote that "I fixed this long ago" I meant only that I had modified the unit test to skip a test unless being run on Linux where it was known to pass. Primarily to allow effort to be focused on problems that were clearly not possibly due to a test relying on a quirk of a particular OS. While investigating bugs identified by ASAN (see #1272) this issue came to my attention again. Using the fix I've written for that ASAN bug the disabled test now passes on macOS.
